.github/workflows/base-library.yml

```diff
@@ -37,21 +37,22 @@ jobs:
       CMAKE_CONFIGURATION: macos_base
     steps:
       - uses: actions/checkout@v2
-      - name: Install Deps
+      - name: build_and_test
         run: |
           set -x
-          brew install ninja openmpi swig openssl python@3.9
-          brew unlink python
-          brew link --force python@3.9
+          brew install openmpi swig openssl
           python3 -mpip install numpy mpi4py
-      - name: Setup CCache
-        uses: hendrikmuhs/ccache-action@v1.2
-        with:
-          key: ${{ github.job }}-macos-ccache
-      - name: Configure
-        run: ctest -VV -S .github/ci/ctest_configure.cmake
-      - name: Build
-        run: ctest -VV -S .github/ci/ctest_build.cmake
-      - name: Test
-        run: ctest -VV -S .github/ci/ctest_test.cmake
-
+          which python3
+          python3 --version
+          which python3-config
+          python3-config --cflags --ldflags
+          python3 -c "import numpy; print(numpy.get_include())"
+          ls $(brew --prefix)/lib/python3.11/site-packages
+          ls $(brew --prefix)/bin
+          mkdir build
+          cd build
+          cmake -DPython_FIND_STRATEGY=LOCATION -DPython_FIND_FRAMEWORK=NEVER -DCMAKE_BUILD_TYPE=Debug -DBUILD_TESTING=ON -DTEST_NP=2 ..
```

Comment:
The purpose of the steps is to make clear what is run isolated from from everything else. It is very helpful when looking at the logs from failures, it will only give you the logs of then event where the failure occurred. The way paging works in the UI having everything in one log is not great.

Reply:
I'm simplifying things to debug why cmake is selecting thew wrong python. when I figure out how to solve that I'll reinstate your cmake scripts